ante/fee.go

```diff
@@ -54,16 +86,52 @@ func (mfd BypassMinFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 				fee := gp.Amount.Mul(glDec)
 				requiredFees[i] = sdk.NewCoin(gp.Denom, fee.Ceil().RoundInt())
 			}
-
-			if !feeCoins.IsAnyGTE(requiredFees) {
+			// 1stake is not a subset of 0stake or 0photon or 1photon, skip the min gas price check
+			// can still the DenomsSubsetOf() from sdk
+			if !feeCoins.DenomsSubsetOf(requiredFees.Sort()) {
+				return next(ctx, tx, simulate)
+			}
+			// requiredFees here is ensured not all zero, when check min_gas_price, fee might be zero. if min_gas_price=0stake,1photon, and feecoins is 0stake, it should not return err. so use IsAnyGTE() rather than IsAnyGTE() from sdk.
+			if !IsAnyGTE(feeCoins, requiredFees) {
 				return ctx, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFee, "insufficient fees; got: %s required: %s", feeCoins, requiredFees)
 			}
 		}
 	}
 
+	// when the tx is bypass msg type, still need to check the denom is not random denom
+	// this is to prevent the situation that a bypass msg carries random fee denoms
+	if ctx.IsCheckTx() && !simulate && (mfd.bypassMinFeeMsgs(msgs) && gas <= uint64(len(msgs))*maxBypassMinFeeMsgGasUsage) && mfd.GlobalMinFee.Has(ctx, types.ParamStoreKeyMinGasPrices) {
```

Comment:
```suggestion
	if ctx.IsCheckTx() && !simulate && allowedToBypassMinFee {
```
why does it check for `mfd.GlobalMinFee.Has(ctx, types.ParamStoreKeyMinGasPrices) ` here?
I read the logic as:
* it's a checkTx
* it's not a simulation
* it's allowed to bypass min fee
* the global bypass parameter is not empty

if the global bypass parameter is empty it should be considered to have the `DefaultZeroGlobalFee` and then the denom should still be checked, right?

Reply:
for bypass msg, if because if we do not have global fee module, then we do not check global fee denom.

Reply:
this is a good point, so the DefaultZeroGlobalFee is not in the store or genesis, it only exists in logics when checking the fee or `getGlobalFee` the fee.

` mfd.GlobalMinFee.Has(ctx, types.ParamStoreKeyMinGasPrices) == true` means we have the global fee module, but the value might be empty, zero, or nonzero.